validator/client/validator.go

```diff
@@ -516,27 +517,27 @@ func (v *validator) UpdateProtections(ctx context.Context, slot uint64) error {
 	if err != nil {
 		return errors.Wrap(err, "could not get attester history")
 	}
-	v.attesterHistoryByPubKeyLock.Lock()
+	newLocks := make(map[[48]byte]*mputil.Lock, len(attestingPubKeys))
+	for _, pubKey := range attestingPubKeys {
+		lock := mputil.NewMultilock(string(pubKey[:]))
+		newLocks[pubKey] = lock
+	}
+	for _, lock := range v.pkLocks {
+		lock.Lock()
+		defer lock.Unlock()
```

Comment:
@rauljordan wdyt?
not the best pattern but the other way is to store oldLocks and iterate through them